### PR TITLE
[CELEBORN-1254][FOLLOWUP] Rename celeborn.worker.sortPartition.reservedMemory.enabled to celeborn.worker.sortPartition.prefetch.enabled

### PR DESCRIPTION
**Pull request by SteNicholas**
### What changes were proposed in this pull request?

Rename `celeborn.worker.sortPartition.reservedMemory.enabled` to `celeborn.worker.sortPartition.prefetch.enabled`. Address [r1469066327](https://github.com/apache/incubator-celeborn/pull/2264/files#r1469066327) of @pan3793.

### Why are the changes needed?

`celeborn.worker.sortPartition.reservedMemory.enabled` is misleading, which should represent that prefetch the original partition files during the first sequential reading path to leverage the Linux PageCache mechanism to speed up the subsequent random reading of them. The config name could use `celeborn.worker.sortPartition.prefetch.enabled` which is is more accurate.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

No.

### CODE REVIEW

**Comment by SteNicholas** [comment]
cc @pan3793, @FMX

File: worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionFilesSorter.java [hunk 1]
```diff
@@ -564,7 +564,7 @@ class FileSorter {
       this.originFilePath = fileInfo.getFilePath();
       this.sortedFilePath = Utils.getSortedFilePath(originFilePath);
       this.isHdfs = fileInfo.isHdfs();
-      this.isWarmUp = !isHdfs && reservedMemoryEnabled;
+      this.isWarmUp = !isHdfs && prefetchEnabled;
```
Comment (pan3793) [inline worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionFilesSorter.java:567]: can we rename isWarmUp to prefetch?

Comment (pan3793) [reply worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionFilesSorter.java:567]: anyway, this is an internal flag, we can change it anytime without bothering users

**Review by pan3793 — APPROVED** [review]
LGTM, if @FMX has no concerns.

**Review by cxzl25 — APPROVED** [review]

**Comment by SteNicholas** [comment]
Merged into main(v0.5.0).

**Pull request closed without merging**